### PR TITLE
Reimplementing AR parser from scratch.

### PR DESCRIPTION
**Pull request by aemartinez**
The goal of this PR is to address #3699 . I was able to implement `fetch_production` from scratch consuming CAMMESA's APIs. I deleted old code for `fetch_price` and `fetch_exchange` since I understand they were not working, this cleans up the code and will make it easier to work on future improvements.

### CODE REVIEW

**Comment by alixunderplatz** [comment]
Hi @aemartinez!
Awesome job and nice clean-up 👍
I just tested the output, looks good to me production-wise, so the fix could go live quickly :)

fetch_production() ->
{'zoneKey': 'AR', 'production': {'biomass': 137.28, 'hydro': 1755.58, 'solar': 0.0, 'wind': 1391.17, 'nuclear': 1460.5, 'unknown': 10062.8}, 'capacity': {}, 'storage': {}, 'source': 'api.cammesa.com', 'datetime': datetime.datetime(2022, 1, 31, 8, 55, tzinfo=tzutc())}

One remark from my side: Your parser is fetching the latest output for the most recent timestamp, which is fine as long as the API on the other side is responding. It is possible/easy to fetch the output for all the available matching 5-min timestamps of the raw JSONs to have the most recent "daily history" of the available data?

**Comment by aemartinez** [comment]
Hi @alixunderplatz ! Thanks for the review. 

Yes, we could fetch the output for all the available matching 5-min timestamps. Should that be done by matching the parameter `target_datetime`? Should we only return the available data if the timestamp from CAMMESA exactly matches `target_datetime` or should we return the closest data?

**Comment by gvillo** [comment]
For the records I sent an email to Cammesa to request some additional information to have a detailed Fossil consumption. We will see how it goes

**Review by gvillo — APPROVED** [review]
LGTM, I don't have write access, if @madsnedergaard could take a look on this one it would be great

**Comment by madsnedergaard** [comment]
As has already been said this is awesome! 💪 

Regarding fetching more data, you can see an example of how we do this in other parsers - for example in [FR.py](https://github.com/electricitymap/electricitymap-contrib/blob/master/parsers/FR.py). If anyone remembers another parser that is more similar or easier to use for inspiration, please do share :)

**Comment by aemartinez** [comment]
Great! Thanks @madsnedergaard for the clarification. After reading other parsers now I see that `fetch_production` can return a `list` of production mixes and not just one. Is that correct? (I hadn't understood that from [this doc](https://github.com/electricitymap/electricitymap-contrib/wiki/Building-a-new-parser#fetch_production).)

If that is the case (`fetch_production` can return a list) then I can work on it so we can have that code in this PR too. Thanks!

File: parsers/AR.py
Comment (aemartinez) [inline parsers/AR.py:44]: Is `.to('UTC')` necessary here? I'm not sure if the `datetime` expects the time to be in a specific timezone.

File: parsers/AR.py
Comment (aemartinez) [inline parsers/AR.py:114]: Should we just put all of the thermal into 'gas' or leave it to 'unknown'? It's a pity to put that into unknown because it accounts for most of the energy production.

**Comment by aemartinez** [comment]
Hi! I managed to return the list of all available timestamps. Please take a look at the last version. I also made two questions about some decisions I had to make in the code. Thanks! 😸

File: parsers/AR.py
Comment (madsnedergaard) [reply parsers/AR.py:40]: It should be in UTC, so this is good :)

File: parsers/AR.py
Comment (madsnedergaard) [reply parsers/AR.py:110]: For now I think it's better to keep it as unknown, even though it would be great to attribute it more appropriately. Would be great to reach out to the source and ask if there's any way they could consider splitting it up in their API :)
(But that's a next step we can do afterwards)

**Comment by madsnedergaard** [comment]
> After reading other parsers now I see that fetch_production can return a list of production mixes and not just one. Is that correct? (I hadn't understood that from [this doc](https://github.com/electricitymap/electricitymap-contrib/wiki/Building-a-new-parser#fetch_production).)

Yes exactly, although I see it's not very clear from the documentation - I have tried updating the wiki to better reflect it now :)

**Review by madsnedergaard — APPROVED** [review]
Awesome! 💪 
Screenshot 2022-02-04 at 07 30 32

**Pull request merged**
